dvc/scm/git/backend/dulwich.py

```diff
@@ -642,3 +644,15 @@ def merge(
         squash: bool = False,
     ) -> Optional[str]:
         raise NotImplementedError
+
+    def validate_git_repo(self, url: str):
+        from dulwich.client import LocalGitClient, get_transport_and_path
+        from dulwich.porcelain import get_remote_repo
+
+        try:
+            _, location = get_remote_repo(self.repo, url)
+            client, path = get_transport_and_path(location)
+        except Exception as exc:
+            raise InvalidRemoteSCMRepo(url) from exc
+        if isinstance(client, LocalGitClient) and not os.path.exists(path):
```

Comment:
```suggestion
        if isinstance(client, LocalGitClient) and not os.path.exists(os.path.join("", path)):
```

We need the `os.path.join("", ...)` here since dulwich assumes `path` is really `file:///path` (and always adds the leading `/` to make it an abspath)